temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -127,6 +128,7 @@ private Result failureToResult(PollWorkflowTaskQueueResponse workflowTask, Throw
           RespondWorkflowTaskCompletedRequest.newBuilder()
               .setTaskToken(workflowTask.getTaskToken())
               .setIdentity(options.getIdentity())
+              .setBinaryChecksum(options.getBinaryChecksum())
```

Comment:
I would make the default null and not call this method if it is not set.

Reply:
I did that initially, but didn't like the extra if statement and decided to add a default.
Should we have an empty string default instead?

Reply:
I've changed it to empty string by default, let me know if you think that's sufficient.